### PR TITLE
Introduce NumberComparison validator

### PR DESCRIPTION
**Pull request by gsteel**
|    Q          |   A
|-------------- | ------
| Documentation | yes
| New Feature   | yes

### Description

Adds a single validator that only validates numeric input with optional lower and upper bounds.

This validator should replace `LessThan`, `GreaterThan` and `Between`

Further discussion on the reason for this validator in #272

### CODE REVIEW

File: docs/book/v2/validators/number-comparison.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Number Comparison Validator
+
+`Laminas\Validator\NumberComparison` allows you to validate if a given value is a numeric value that is either
```
Comment (froschdesign) [inline docs/book/v2/validators/number-comparison.md]:
```suggestion
`Laminas\Validator\NumberComparison` allows you to validate if a given value is a numeric value that is either:
```

File: docs/book/v2/validators/number-comparison.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+- Between a min and max value
+- Greater than a min value
+- Less than a max value
+
+By default, comparisons are inclusive. 
+
+CAUTION: **Only supports number validation**
+`Laminas\Validator\NumberComparison` supports only the validation of numbers. Strings or dates can not be validated with this validator.
```
Comment (froschdesign) [inline docs/book/v2/validators/number-comparison.md]:
```suggestion
`Laminas\Validator\NumberComparison` supports only the validation of numbers.
Strings or dates can not be validated with this validator.
```

File: docs/book/v2/validators/number-comparison.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+```php
+$valid  = new Laminas\Validator\NumberComparison(['min' => 0, 'max' => 10]);
+$value  = 10;
+$result = $valid->isValid($value);
+// returns true
+```
+
+In the above example, the result is `true` due to the reason that the default search is inclusive of the border values.
+This means in our case that any value from '0' to '10' is allowed; values like '-1' and '11' will return `false`.
```
Comment (froschdesign) [inline docs/book/v2/validators/number-comparison.md]:
```suggestion
This means in our case that any value from `0` to `10` is allowed; values like `-1` and `11` will return `false`.
```

File: docs/book/v2/validators/number-comparison.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    'inclusiveMin' => false,
+    'inclusiveMax' => false,
+]);
+
+$valid->isValid(10); // false
+$valid->isValid(0);  // false
+$valid->isValid(9);  // true
+```
+
+The example above is almost identical to our first example, but we now exclude the bounds as valid values; as such, the values '0' and '10' are no longer allowed and will return `false`.
```
Comment (froschdesign) [inline docs/book/v2/validators/number-comparison.md]:
```suggestion
The example above is almost identical to our first example, but we now exclude the bounds as valid values; as such, the values `0` and `10` are no longer allowed and will return `false`.
```

File: docs/book/v2/validators/number-comparison.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+The following options are supported for `Laminas\Validator\NumberComparison`:
+
+| Option         | Data Type | Default Value | Description                                                              |
+|----------------|-----------|---------------|--------------------------------------------------------------------------|
+| `max`          | `numeric` | `null`        | Sets the upper bound for the input.                                      |
+| `min`          | `numeric` | `null`        | Sets the lower bound for the input.                                      |
+| `inclusiveMin` | `bool`    | `true`        | Defines if the validation is inclusive of the lower bound, or exclusive. |
+| `inclusiveMax` | `bool`    | `true`        | Defines if the validation is inclusive of the upper bound, or exclusive. |
+
+## Default Behaviour
```
Comment (froschdesign) [inline docs/book/v2/validators/number-comparison.md]:
```suggestion
## Basic Usage
```

**Review by froschdesign — CHANGES_REQUESTED** [review]
Only minor formatting changes and a heading.

**Review by froschdesign — APPROVED** [review]
LGTM 👍🏻

**Review by Ocramius — APPROVED** [review]
:ship: thanks @gsteel!

**Pull request merged**